[illumos] `TestConnectedSocket.test_unix` fails

## Summary

* OS: OpenIndiana (illumos)
* Architecture: 64bit
* Psutil version: 7.0.0
* Python version: Python 3.9.21
* Type: core

## Description

The following test fails on OpenIndiana:
```
________________________ TestConnectedSocket.test_unix _________________________

self = <psutil.tests.test_connections.TestConnectedSocket testMethod=test_unix>

    @pytest.mark.skipif(not POSIX, reason="POSIX only")
    def test_unix(self):
        testfn = self.get_testfn()
        server, client = unix_socketpair(testfn)
        try:
>           cons = this_proc_net_connections(kind='unix')

psutil/tests/test_connections.py:232:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
psutil/tests/test_connections.py:50: in this_proc_net_connections
    cons = psutil.Process().net_connections(kind=kind)
psutil/__init__.py:1234: in net_connections
    _check_conn_kind(kind)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

kind = 'unix'

    def _check_conn_kind(kind):
        """Check net_connections()'s `kind` parameter."""
        kinds = tuple(_common.conn_tmap)
        if kind not in kinds:
            msg = f"invalid kind argument {kind!r}; valid ones are: {kinds}"
>           raise ValueError(msg)
E           ValueError: invalid kind argument 'unix'; valid ones are: ('all', 'tcp', 'tcp4', 'udp', 'udp4', 'inet', 'inet4', 'inet6', 'tcp6', 'udp6')

psutil/__init__.py:280: ValueError
```
When I revert (partially) changeset 5d4be42bd309dc3c7e93d5d319577ec4b8027445 using this:
```
--- psutil-7.0.0/psutil/_common.py.orig
+++ psutil-7.0.0/psutil/_common.py
@@ -249,7 +249,7 @@
         "udp6": ([AF_INET6], [SOCK_DGRAM]),
     })

-if AF_UNIX is not None and not SUNOS:
+if AF_UNIX is not None:
     conn_tmap.update({"unix": ([AF_UNIX], [SOCK_STREAM, SOCK_DGRAM])})


```
Then the test passes.

Actually, there are three more tests that fails similarly:
```
______________________ TestNetUtils.test_unix_socketpair _______________________

self = <psutil.tests.test_testutils.TestNetUtils testMethod=test_unix_socketpair>

    @pytest.mark.skipif(not POSIX, reason="POSIX only")
    @pytest.mark.skipif(
        NETBSD or FREEBSD, reason="/var/run/log UNIX socket opened by default"
    )
    def test_unix_socketpair(self): 
        p = psutil.Process()
        num_fds = p.num_fds()
>       assert not filter_proc_net_connections(p.net_connections(kind='unix'))

psutil/tests/test_testutils.py:331: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
psutil/__init__.py:1234: in net_connections 
    _check_conn_kind(kind)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

kind = 'unix'

    def _check_conn_kind(kind):
        """Check net_connections()'s `kind` parameter."""
        kinds = tuple(_common.conn_tmap)
        if kind not in kinds:
            msg = f"invalid kind argument {kind!r}; valid ones are: {kinds}"
>           raise ValueError(msg)
E           ValueError: invalid kind argument 'unix'; valid ones are: ('all', 'tcp', 'tcp4', 'udp', 'udp4', 'inet', 'inet4', 'inet6', 'tcp6', 'udp6')

psutil/__init__.py:280: ValueError
_____________________ TestFSAPIs.test_proc_net_connections _____________________

self = <psutil.tests.test_unicode.TestFSAPIs testMethod=test_proc_net_connections>

    @pytest.mark.skipif(not POSIX, reason="POSIX only") 
    def test_proc_net_connections(self):
        name = self.get_testfn(suffix=self.funky_suffix)
        sock = bind_unix_socket(name) 
        with closing(sock):
>           conn = psutil.Process().net_connections('unix')[0]

psutil/tests/test_unicode.py:232:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
psutil/__init__.py:1234: in net_connections
    _check_conn_kind(kind)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

kind = 'unix'

    def _check_conn_kind(kind):
        """Check net_connections()'s `kind` parameter."""
        kinds = tuple(_common.conn_tmap)
        if kind not in kinds:
            msg = f"invalid kind argument {kind!r}; valid ones are: {kinds}"
>           raise ValueError(msg)
E           ValueError: invalid kind argument 'unix'; valid ones are: ('all', 'tcp', 'tcp4', 'udp', 'udp4', 'inet', 'inet4', 'inet6', 'tcp6', 'udp6')

psutil/__init__.py:280: ValueError
_____________ TestFSAPIsWithInvalidPath.test_proc_net_connections ______________

self = <psutil.tests.test_unicode.TestFSAPIsWithInvalidPath testMethod=test_proc_net_connections>

    @pytest.mark.skipif(not POSIX, reason="POSIX only") 
    def test_proc_net_connections(self):
        name = self.get_testfn(suffix=self.funky_suffix)
        sock = bind_unix_socket(name) 
        with closing(sock):
>           conn = psutil.Process().net_connections('unix')[0]

psutil/tests/test_unicode.py:232:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
psutil/__init__.py:1234: in net_connections
    _check_conn_kind(kind)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

kind = 'unix'

    def _check_conn_kind(kind):
        """Check net_connections()'s `kind` parameter."""
        kinds = tuple(_common.conn_tmap)
        if kind not in kinds:
            msg = f"invalid kind argument {kind!r}; valid ones are: {kinds}"
>           raise ValueError(msg)
E           ValueError: invalid kind argument 'unix'; valid ones are: ('all', 'tcp', 'tcp4', 'udp', 'udp4', 'inet', 'inet4', 'inet6', 'tcp6', 'udp6')

psutil/__init__.py:280: ValueError
```

--- a/psutil/tests/test_connections.py
+++ b/psutil/tests/test_connections.py
@@ -224,6 +224,9 @@ class TestConnectedSocket(ConnectionTestCase):
             client.close()
 
     @pytest.mark.skipif(not POSIX, reason="POSIX only")
+    @pytest.mark.skipif(
+        not HAS_NET_CONNECTIONS_UNIX, reason="can't list UNIX sockets"
+    )
     def test_unix(self):
         testfn = self.get_testfn()
         server, client = unix_socketpair(testfn)

--- a/psutil/tests/test_testutils.py
+++ b/psutil/tests/test_testutils.py
@@ -324,6 +324,9 @@ class TestNetUtils(PsutilTestCase):
     @pytest.mark.skipif(
         NETBSD or FREEBSD, reason="/var/run/log UNIX socket opened by default"
     )
+    @pytest.mark.skipif(
+        not HAS_NET_CONNECTIONS_UNIX, reason="can't list UNIX sockets"
+    )
     def test_unix_socketpair(self):
         p = psutil.Process()
         num_fds = p.num_fds()

--- a/psutil/tests/test_unicode.py
+++ b/psutil/tests/test_unicode.py
@@ -226,6 +226,9 @@ class TestFSAPIs(BaseUnicodeTest):
             assert os.path.normcase(path) == os.path.normcase(self.funky_name)
 
     @pytest.mark.skipif(not POSIX, reason="POSIX only")
+    @pytest.mark.skipif(
+        not HAS_NET_CONNECTIONS_UNIX, reason="can't list UNIX sockets"
+    )
     def test_proc_net_connections(self):
         name = self.get_testfn(suffix=self.funky_suffix)
         sock = bind_unix_socket(name)

